frontend/src/employee-frontend/components/messages/MessageEditor.tsx

```diff
@@ -159,14 +159,21 @@ export default React.memo(function MessageEditor({
     function updateSelectedReceiversOnReceiverTreeChanges() {
       const selected = getSelected(receiverTree)
       const recipientAccountIds = getSelectedBottomElements(receiverTree)
-      setMessage((old) => ({
-        ...old,
-        recipientAccountIds,
-        recipientIds: selected.map((s) => s.value),
-        recipientNames: selected.map((s) => s.label)
-      }))
+      if (
+        selected
+          .map((s) => s.value)
+          .sort()
+          .toString() !== message.recipientIds.sort().toString()
```

Comment:
`.sort()` mutates the array in place. Are you 100% sure this is ok here? Mutating `selected.map((s) => s.value)` is ok, since we know calling map creates a new array, but it's not obvious to me if it's ok to mutate `message.recipientIds`.

Also, consider using something like lodash `isEqual` to compare arrays instead of comparing their string representations